src/box/tuple_format.c

```diff
@@ -1416,3 +1441,159 @@ tuple_format_iterator_next(struct tuple_format_iterator *it,
 	entry->data = NULL;
 	return 0;
 }
+
+/**
+ * A builder that helps to construct an array from chunks with MsgPack objects.
+ *
+ * First, chunks of data are added to a builder object. The builder doesn't
+ * allocate any memory for the MsgPack, and doesn't copy it, only pointers to
+ * the start and to the end of the data are preserved.
+ *
+ * Once all chunks have been added, the builder can be used to encode them into
+ * the final MsgPack array.
+ */
+struct mp_array_builder {
```

Comment:
We discussed it offline with @locker.
I believe that this utility will be useful later.
Please move it to a separate file(s) and rename it to `tuple_builder` (there's already an `mp_array` essence and that leads to name collision).
Please provide constructor that takes and saves `struct region *` argument for further allocations.
Unit test will be highly appreciated to. Also that deserves a separate commit.

Reply:
Done